types/es-abstract/2015/HasOwnProperty.d.ts

```diff
@@ -0,0 +1,4 @@
+import type { PropertyKey } from '../index';
+
+declare function HasOwnProperty(O: object, P: PropertyKey): boolean;
```

Comment:
is there a way to have this assert that `P extends keyof O`, instead of just returning a boolean?

Reply:
this isn't resolved until there's at least an answer to the question - why not:
```suggest
declare function HasOwnProperty(O: object, P: PropertyKey): P is keyof O;
```